rest_framework/mixins.py

```diff
@@ -69,13 +69,14 @@ def update(self, request, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         self.perform_update(serializer)
 
-        queryset = self.filter_queryset(self.get_queryset())
-        if queryset._prefetch_related_lookups:
+        if hasattr(instance, '_prefetched_objects_cache'):
```

Comment:
When only `get_object` is defined, I assume that folks are unlikely to have advanced prefetches in their method (although it's very much possible). 

If some things were prefetched, we assume that there is a `get_queryset` method.